libmamba/tests/test_data.hpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2022, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#ifndef MAMBA_CORE_URL_HPP
+#define MAMBA_CORE_URL_HPP
+
+#include "mamba/core/fsutil.hpp"
+
+namespace mamba
+{
+
+#ifndef MAMBA_TEST_DATA_DIR
+#error "MAMBA_TEST_DATA_DIR must be defined pointing to test data"
+#endif
+    static const fs::u8path test_data_dir = MAMBA_TEST_DATA_DIR;
```

Comment:
To avoid ODR violation (or at least duplication in multiple object files) I think you can make this object `inline`, so that there is only one definition after link. Either that or mark it extern and put the definition in a cpp). But `inline` should be enough.